app/internal/window/os_js.go

```diff
@@ -417,22 +418,17 @@ func (w *window) funcOf(f func(this js.Value, args []js.Value) interface{}) js.F
 func (w *window) animCallback() {
 	w.mu.Lock()
 	anim := w.animating
-	if anim {
-		w.requestAnimationFrame.Invoke(w.redraw)
-	}
 	w.mu.Unlock()
 	if anim {
 		w.draw(false)
 	}
+	w.requestAnimationFrame.Invoke(w.redraw)
```

Comment:
This is called unconditionally now. When will animation stop?

Reply:
The `if anim {}` is responsable to prevent the redraw. The "animation callback" will never stop. The callback will be triggred each ~16ms (the refresh-rate of the monitor). However, it will not draw a new frame if it's not animating. 

The old way of calling `requestAnimationFrame` only when animate, triggers the callback faster than the refresh-rate.  I think it happens because `SetAnimation` can be true and false in one frame. Or, it can change one frame to another, which might call `requestAnimationFrame` inside the SetAnimation AND calls `requestAnimationFrame` in the current callback, so there's two calls in the same frame (or more).

Reply:
I don't think it's a good idea to always have requestAnimationFrame running, even if it is nearly a no-op. There must be some other way of fixing the piled up requests. How about tracking whether there is a pending requestAnimationFrame, and not submit another if so?

Reply:
First, I did some:
 ```
 t := time.Now()
	for e := range w.Events() {
		switch e := e.(type) {
		case system.FrameEvent:
			gtx := layout.NewContext(&ops, e)
			e.Frame(gtx.Ops)
			fmt.Println("FRAME", t.Sub(time.Now()))
                        t = time.Now()
			w.Invalidate()
```
The `fmt.Println` prints the time betwwen each frame. On Windows/Android it's 16ms. On WASM it's 6ms (or 2ms with #4 applied). Anyway, the WASM is creating frame faster than intended! 

I found that the `SetAnimating` calls the `w.requestAnimationFrame` and also it's called inside the callback, so I guess... It's calling twice. I insert some `fmt.Println`, the `w.SetAnimating` sometimes is `false` and them `true` (I don't know if it's one the same frame, or not), but... Since it's false > true, it could call `w.requestAnimationFrame` again, and again.

_If you test the same code (above) with that PR, the timing will be ~16ms._

Reply:
I'm sure the PR fixes what it sets out to do (16 ms) frame, but I'm saying it's not worth the cost: always having requestAnimationFrame callbacks, even if the program is otherwise idle.

I suggested tracking whether a requestAnimationFrame has been called, and only submitting a new one if not. I believe that will achieve the same effect: reducing the updates to once every ~16ms, but without callbacks while idle (not animating).

Reply:
I don't know to better fix that. In theory the old code was suppose to prevent multiples `w.requestAnimationFrame.Invoke`:
```
if anim && !w.animating {	
		w.requestAnimationFrame.Invoke(w.redraw)
``` 

But, doesn't seems to prevent such duplication. The another option might be adding some loop/select, but I think it's worse than call  `w.requestAnimationFrame.Invoke` everytime.